In [2]:
import pandas as pd
import numpy as np

In [3]:
df = pd.read_csv('/workspaces/Final_Project_DataScient/data/processed/data_1_selected_39.csv')
df

HHADULT  SEXVAR  MEDCOST1  SLEPTIM1  CVDSTRK3  ADDEPEV3  DIABETE4  \
0           2.0     2.0       2.0       6.0       2.0       1.0       3.0   
1           4.0     2.0       2.0       7.0       2.0       2.0       3.0   
2           2.0     2.0       2.0       7.0       2.0       2.0       3.0   
3           2.0     1.0       2.0       6.0       2.0       2.0       1.0   
4           2.0     1.0       2.0       5.0       2.0       2.0       3.0   
...         ...     ...       ...       ...       ...       ...       ...   
110375      1.0     1.0       2.0       6.0       2.0       2.0       3.0   
110376      4.0     1.0       2.0       7.0       1.0       2.0       1.0   
110377      2.0     2.0       2.0       7.0       2.0       2.0       3.0   
110378      1.0     2.0       1.0       5.0       2.0       2.0       3.0   
110379      2.0     1.0       2.0       5.0       2.0       2.0       3.0   

        MARITAL  RENTHOM1  VETERAN3  ...  _ASTHMS1  _DRDXAR2  _AGEG5YR  \
0           1.0       1.0       2.0  ...       3.0       2.0       9.0   
1           1.0       1.0       2.0  ...       3.0       2.0       9.0   
2           1.0       2.0       2.0  ...       3.0       1.0       7.0   
3           1.0       1.0       2.0  ...       3.0       2.0       7.0   
4           1.0       1.0       1.0  ...       3.0       2.0       9.0   
...         ...       ...       ...  ...       ...       ...       ...   
110375      2.0       1.0       2.0  ...       3.0       1.0       9.0   
110376      1.0       2.0       2.0  ...       3.0       1.0      10.0   
110377      1.0       1.0       1.0  ...       3.0       2.0       7.0   
110378      1.0       3.0       2.0  ...       3.0       2.0      10.0   
110379      1.0       2.0       2.0  ...       2.0       2.0      11.0   

        _BMI5CAT  _CHLDCNT  _EDUCAG  _INCOMG1  _RFBING6  _AIDTST4  tobacco_use  
0            2.0       1.0      4.0       5.0       1.0       2.0          0.0  
1            3.0       1.0      4.0       9.0       1.0       2.0          0.0  
2            4.0       1.0      2.0       3.0       1.0       2.0          0.0  
3            4.0       1.0      3.0       6.0       1.0       2.0          0.0  
4            4.0       1.0      3.0       4.0       1.0       1.0          0.0  
...          ...       ...      ...       ...       ...       ...          ...  
110375       3.0       1.0      3.0       6.0       2.0       2.0          1.0  
110376       3.0       1.0      4.0       6.0       1.0       1.0          0.0  
110377       3.0       2.0      4.0       5.0       1.0       1.0          0.0  
110378       1.0       1.0      2.0       9.0       9.0       2.0          2.0  
110379       4.0       1.0      3.0       5.0       1.0       1.0          0.0  

[110380 rows x 39 columns]

In [4]:
df.columns

Index(['HHADULT', 'SEXVAR', 'MEDCOST1', 'SLEPTIM1', 'CVDSTRK3', 'ADDEPEV3',
       'DIABETE4', 'MARITAL', 'RENTHOM1', 'VETERAN3', 'EMPLOY1', 'DECIDE',
       'DIFFALON', 'COLNCNCR', 'HIVRISK5', 'COVIDPOS', 'LSATISFY', 'EMTSUPRT',
       'SDHISOLT', 'SDHEMPLY', 'SDHFOOD1', 'SDHBILLS', 'SDHUTILS', 'SDHTRNSP',
       '_PHYS14D', '_MENT14D', '_HLTHPLN', '_TOTINDA', '_MICHD', '_ASTHMS1',
       '_DRDXAR2', '_AGEG5YR', '_BMI5CAT', '_CHLDCNT', '_EDUCAG', '_INCOMG1',
       '_RFBING6', '_AIDTST4', 'tobacco_use'],
      dtype='object')

ALCOHOL

In [5]:
# 1: No, 2: Yes >> Cambiar a 1: Yes, 0: No
df.loc[df["_RFBING6"] > 2, "_RFBING6"] = -1
df.loc[df["_RFBING6"] == 1, "_RFBING6"] = 0
df.loc[df["_RFBING6"] == 2, "_RFBING6"] = 1
df._RFBING6.value_counts()

_RFBING6
 0.0    95518
 1.0    12883
-1.0     1979
Name: count, dtype: int64

0.0 → No bebedor compulsivo : 95,518 personas.

1.0 → Bebedor compulsivo: 12,883 personas. 

-1.0 → Falta de respuesta / dato perdido / desconocido: 1,979 personas.

AISLAMIENTO SOCIAL

In [6]:
# 1, 2 y 3: Aislado, 4 y 5: No aislado >> Cambiar a 1: Aislado, 0: No aislado
df.loc[df["SDHISOLT"] <= 3, "SDHISOLT"] = 1
df.loc[(df["SDHISOLT"] == 4) | (df["SDHISOLT"] == 5), "SDHISOLT"] = 0
df.loc[df["SDHISOLT"] > 5, "SDHISOLT"] = -1
df["SDHISOLT"].value_counts()

SDHISOLT
 0.0    80904
 1.0    28607
-1.0      869
Name: count, dtype: int64

0.0 : 72.2%  → No presenta aislamiento social.

1.0 : 25.5% → Sí presenta aislamiento social.

-1.0 : 0.8% → Dato no disponible / no respondió / desconocido.

Hay una proporción considerable de personas con aislamiento social una de cada cuatro, es una variable predictora muy a tener en cuenta.


APOYO SOCIAL Y EMOCIONAL

In [27]:
df.loc[df["EMTSUPRT"] <= 2, "EMTSUPRT"] = 1
df.loc[(df["EMTSUPRT"] == 3) | (df["EMTSUPRT"] == 4) | (df["EMTSUPRT"] == 5), "EMTSUPRT"] = 0
df.loc[df["EMTSUPRT"] > 5, "EMTSUPRT"] = -1
df["EMTSUPRT"].value_counts()

EMTSUPRT
 1.0    87214
 0.0    21615
-1.0     1551
Name: count, dtype: int64

1.0 → Sí recibe apoyo social/emocional: 87,214 personas.

0.0 → No recibe ese tipo de apoyo: 21,615 personas.

-1.0 → Dato faltante / sin respuesta / desconocido: 1,551 personas.

Más de tres cuartas partes reportan tener apoyo emocional. Esto puede ser un factor protector, el 20% sin apoyo emocional puede ser crítico, este grupo minoritario puede tener mayor riesgo de aislamiento, ansiedad, depresión o conductas nocivas.



DEPRESION

In [28]:
# Lo llevamos a una variable binaria.
# Eliminamos NAN y los valores 7 y 9 (Don’t know/Refused/Missing)
# Pasamos 2 (no) a 0
df['ADDEPEV3'].dropna()
df = df[~df["ADDEPEV3"].isin([9,7])]
df.loc[df["ADDEPEV3"] == 2, "ADDEPEV3"] = 0
print(df["ADDEPEV3"].value_counts())
print(df.shape)

ADDEPEV3
0.0    87246
1.0    22740
Name: count, dtype: int64
(109986, 39)


/tmp/ipykernel_5781/1116977196.py:6: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df.loc[df["ADDEPEV3"] == 2, "ADDEPEV3"] = 0


EXAMEN DE COLON

In [29]:
# Examen de colon - ¿Te lo haz hecho?
df.loc[df["COLNCNCR"] == 2, "COLNCNCR"] = 0 #no
df.loc[df["COLNCNCR"] > 5, "COLNCNCR"] = -1 #no sabe
# Ver distribución de valores
print(df["COLNCNCR"].value_counts())

COLNCNCR
 0.0    84062
 1.0    23275
-1.0     2649
Name: count, dtype: int64


/tmp/ipykernel_5781/568849800.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df.loc[df["COLNCNCR"] == 2, "COLNCNCR"] = 0 #no
/tmp/ipykernel_5781/568849800.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df.loc[df["COLNCNCR"] > 5, "COLNCNCR"] = -1 #no sabe


1.0 → Sí se ha hecho el examen.

0.0 → No se lo ha hecho.

-1.0 → No sabe / sin respuesta. 



ENFERMEDAD CARDIACA

In [30]:
# ¿Haz tenido una enfermada del corazon? 
df.loc[df["_MICHD"] == 2, "_MICHD"] = 0 #no

# Ver distribución de valores
print(df["_MICHD"].value_counts())

_MICHD
0.0    97657
1.0    12329
Name: count, dtype: int64


/tmp/ipykernel_5781/3541356943.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df.loc[df["_MICHD"] == 2, "_MICHD"] = 0 #no


0.0 → No ha tenido enfermedad cardíaca: 97,657 personas

1.0 → Sí ha tenido enfermedad cardíaca: 12,329 personas

Esto nos indica que alrededor del 11% de la población ha reportado enfermedad cardíaca. Esto es coherente con cifras reales de muchas poblaciones donde los problemas cardiovasculares son una de las principales causas de enfermedad crónica, especialmente en personas mayores.


ESTADO DE ASMA 

In [31]:
# ¿haz tenido asma?
df.loc[df["_ASTHMS1"] == 9, "_ASTHMS1"] = -1 #missing

# Ver distribución de valores
print(df["_ASTHMS1"].value_counts())

_ASTHMS1
 3.0    93717
 1.0    11473
 2.0     4069
-1.0      727
Name: count, dtype: int64


/tmp/ipykernel_5781/2159376363.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df.loc[df["_ASTHMS1"] == 9, "_ASTHMS1"] = -1 #missing


1.0 → Actualmente tiene asma.

2.0 → Tuvo asma en algún momento, pero no actualmente.

3.0 → Nunca ha tenido asma

-1.0 → Sin respuesta / no sabe / dato faltante

Alrededor del 13% de la población ha tenido asma en algún momento. El 9.6% aún la padece, lo que puede representar un grupo con necesidades especiales de atención o prevención. El 78.7% nunca ha tenido asma, lo que podría servir como grupo de referencia o comparación.

ENFERMEDAD ARTRITIS

In [32]:
df.loc[df["_DRDXAR2"] == 2, "_DRDXAR2"] = 0 #no
# Ver distribución de valores
print(df["_DRDXAR2"].value_counts())

_DRDXAR2
0.0    63609
1.0    46377
Name: count, dtype: int64


/tmp/ipykernel_5781/3334935505.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df.loc[df["_DRDXAR2"] == 2, "_DRDXAR2"] = 0 #no


0.0 → No ha sido diagnosticada con artritis. 57.8%

1.0 → Sí ha recibido diagnóstico de artritis. 42.2%

Estos datos nos dice que 4 de cada 10 personas declaran tener diagnóstico de artritis, lo cual es altísimo y refleja la carga de enfermedades musculoesqueléticas, sobre todo en poblaciones adultas y mayores. **Perspectiva poblacional:** En estudios reales, prevalencias entre 15% y 30% son comunes — si aquí es 42%, eso puede indicar una muestra envejecida o con más factores de riesgo acumulados.

MALA SALUD FISICA

In [33]:
# Mala salud fisica de 0 (no), de 1 a 30 dias (si)
df.loc[df["_PHYS14D"] == 1, "_PHYS14D"] = 0 #no
df.loc[df["_PHYS14D"] == 2, "_PHYS14D"] = 1 #-13dias
df.loc[df["_PHYS14D"] == 3, "_PHYS14D"] = 2 #+13 dias
df.loc[df["_PHYS14D"] == 9, "_PHYS14D"] = -1 #missing

# Ver distribución de valores
print(df["_PHYS14D"].value_counts())

_PHYS14D
 0.0    65609
 1.0    25744
 2.0    16354
-1.0     2279
Name: count, dtype: int64


/tmp/ipykernel_5781/3638456170.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df.loc[df["_PHYS14D"] == 1, "_PHYS14D"] = 0 #no
/tmp/ipykernel_5781/3638456170.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df.loc[df["_PHYS14D"] == 2, "_PHYS14D"] = 1 #-13dias
/tmp/ipykernel_5781/3638456170.py:4: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df.loc[df["_PHYS14D"] == 3, "_PHYS14D"] = 2 #+13 dias
/tmp/ipykernel_5781/3638456170.py:5

0.0 → No experimentó mala salud física

1.0 → 1 a 13 días de mala salud física

2.0 → Los 14 días completos de mala salud física

-1.0 → Sin datos / no responde

Casi un tercio de la muestra (35%) experimentó mala salud física en los últimos 14 días, ya sea de forma intermitente (21.6%) o diaria (13.7%).

Más de la mitad (55%) reporta estar sin problemas físicos recientes.

MALA SALUD MENTAL

In [34]:
# Mala salud metal de 0, (no) a 30 dias (si)
df.loc[df["_MENT14D"] == 1, "_MENT14D"] = 0 #0 dia
df.loc[df["_MENT14D"] == 2, "_MENT14D"] = 1 #-13 dias
df.loc[df["_MENT14D"] == 3, "_MENT14D"] = 2 #+13 dias
df.loc[df["_MENT14D"] == 9, "_MENT14D"] = -1 #missing
print(df["_MENT14D"].value_counts())

_MENT14D
 0.0    71693
 1.0    23982
 2.0    12488
-1.0     1823
Name: count, dtype: int64


/tmp/ipykernel_5781/2202803867.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df.loc[df["_MENT14D"] == 1, "_MENT14D"] = 0 #0 dia
/tmp/ipykernel_5781/2202803867.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df.loc[df["_MENT14D"] == 2, "_MENT14D"] = 1 #-13 dias
/tmp/ipykernel_5781/2202803867.py:4: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df.loc[df["_MENT14D"] == 3, "_MENT14D"] = 2 #+13 dias
/tmp/ipykernel_5781/2202803867.

0.0 → Sin días de malestar mental

1.0 → Entre 1 y 13 días de mala salud mental

2.0 → Todos los 14 días con mal estado mental

-1.0 → Sin respuesta / dato faltante

Alrededor del 30% de la muestra ha experimentado algún nivel de malestar mental reciente.

El 10.4% reporta un nivel grave y sostenido (14 días continuos), lo cual puede indicar síntomas compatibles con depresión o ansiedad severa.

PESO CORPORAL 

In [35]:
# Indice de peso corporal
df.loc[df["_BMI5CAT"] == 1, "_BMI5CAT"] = 0 #bajo peso
df.loc[df["_BMI5CAT"] == 2, "_BMI5CAT"] = 1 #peso normal
df.loc[df["_BMI5CAT"] == 3, "_BMI5CAT"] = 2 #sobrepeso
df.loc[df["_BMI5CAT"] == 4, "_BMI5CAT"] = 3 #obeso
df.loc[df["_BMI5CAT"] == 9, "_BMI5CAT"] = -1 #missing
print(df["_BMI5CAT"].value_counts())

_BMI5CAT
2.0    40997
3.0    39131
1.0    28532
0.0     1326
Name: count, dtype: int64


/tmp/ipykernel_5781/3315613289.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df.loc[df["_BMI5CAT"] == 1, "_BMI5CAT"] = 0 #bajo peso
/tmp/ipykernel_5781/3315613289.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df.loc[df["_BMI5CAT"] == 2, "_BMI5CAT"] = 1 #peso normal
/tmp/ipykernel_5781/3315613289.py:4: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df.loc[df["_BMI5CAT"] == 3, "_BMI5CAT"] = 2 #sobrepeso
/tmp/ipykernel_5781/331

0.0 → Peso bajo o bajo peso (IMC < 18.5)

1.0 → Peso normal (IMC 18.5–24.9)

2.0 → Sobrepeso (IMC 25.0–29.9)

3.0 → Obesidad (IMC ≥ 30)

Más de 2 de cada 3 personas tienen sobrepeso u obesidad (~67.2%), lo cual es un hallazgo clínico y poblacional muy importante.

DIFICULTAD PARA CONCENTRARSE

In [36]:
# Dificultad para concentrarse
df.loc[df["DECIDE"] == 2, "DECIDE"] = 0 #no
df.loc[df["DECIDE"] > 5, "DECIDE"] = -1 #missing
print(df["DECIDE"].value_counts())

DECIDE
 0.0    97716
 1.0    11692
-1.0      578
Name: count, dtype: int64


/tmp/ipykernel_5781/1148006535.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df.loc[df["DECIDE"] == 2, "DECIDE"] = 0 #no
/tmp/ipykernel_5781/1148006535.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df.loc[df["DECIDE"] > 5, "DECIDE"] = -1 #missing


0.0 → No reporta dificultad para concentrarse o decidir

1.0 → Sí reporta dificultad cognitiva

-1.0 → No sabe / No responde / Dato faltante

Un 10.4% de las personas presenta dificultad cognitiva, lo cual es relevante desde una perspectiva de salud pública.

DIFICULTAD PARA HACER ACTIVIDAD FISICA 

In [37]:
# Dificultad para hacer alguna activida fisica (lesion)
df.loc[df["DIFFALON"] == 2, "DIFFALON"] = 0 #no
df.loc[df["DIFFALON"] > 5, "DIFFALON"] = -1 #missing
print(df["DIFFALON"].value_counts())

DIFFALON
 0.0    101687
 1.0      8057
-1.0       242
Name: count, dtype: int64


/tmp/ipykernel_5781/2484800539.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df.loc[df["DIFFALON"] == 2, "DIFFALON"] = 0 #no
/tmp/ipykernel_5781/2484800539.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df.loc[df["DIFFALON"] > 5, "DIFFALON"] = -1 #missing


0.0 → No tiene dificultad para hacer actividad física

1.0 → Sí tiene dificultad para hacer actividad física

-1.0 → No sabe / No responde / Dato faltante

TEST DE COVID POSITIVO

In [38]:
df.loc[df["COVIDPOS"] == 2, "COVIDPOS"] = 0 #no
df.loc[df["COVIDPOS"] == 1, "COVIDPOS"] = 2 #no
df.loc[df["COVIDPOS"] == 3, "COVIDPOS"] = 1 #no
df.loc[df["COVIDPOS"] > 3, "COVIDPOS"] = -1 #missing
print(df["COVIDPOS"].value_counts())

/tmp/ipykernel_5781/2426322530.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df.loc[df["COVIDPOS"] == 2, "COVIDPOS"] = 0 #no
/tmp/ipykernel_5781/2426322530.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df.loc[df["COVIDPOS"] == 1, "COVIDPOS"] = 2 #no


COVIDPOS
 0.0    75906
 2.0    29761
 1.0     4063
-1.0      256
Name: count, dtype: int64


/tmp/ipykernel_5781/2426322530.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df.loc[df["COVIDPOS"] == 3, "COVIDPOS"] = 1 #no
/tmp/ipykernel_5781/2426322530.py:4: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df.loc[df["COVIDPOS"] > 3, "COVIDPOS"] = -1 #missing


0.0 → Nunca se ha hecho una prueba de COVID-19

1.0 → Se hizo la prueba y dio positivo

2.0 → Se hizo la prueba y dio negativo

-1.0 → No sabe / no responde / dato faltante

Casi 2 de cada 3 personas nunca se hicieron una prueba, lo cual puede reflejar:
Falta de acceso a testeo, asintomáticos o baja percepción de riesgo.

Solo el 3.4% reporta haber dado positivo, lo cual puede estar subestimado si muchas personas no se testearon.

El 24.8% dio negativo, lo que indica que al menos una cuarta parte de la muestra sí accedió a testeo preventivo o por sospecha.

SATISFACION DE LA VIDA 

In [39]:
# satisfacion con la vida
df.loc[df["LSATISFY"] == 1, "LSATISFY"] = 0 #muy satisfecho
df.loc[df["LSATISFY"] == 2, "LSATISFY"] = 1 #satisfecho
df.loc[df["LSATISFY"] == 3, "LSATISFY"] = 2 #insatisfecho
df.loc[df["LSATISFY"] == 4, "LSATISFY"] = 3 #muy insatisfecho
df.loc[df["LSATISFY"] > 5, "LSATISFY"] = -1 #missing
print(df["LSATISFY"].value_counts())

/tmp/ipykernel_5781/2712223949.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df.loc[df["LSATISFY"] == 1, "LSATISFY"] = 0 #muy satisfecho
/tmp/ipykernel_5781/2712223949.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df.loc[df["LSATISFY"] == 2, "LSATISFY"] = 1 #satisfecho
/tmp/ipykernel_5781/2712223949.py:4: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df.loc[df["LSATISFY"] == 3, "LSATISFY"] = 2 #insatisfecho
/tmp/ipykernel_5

LSATISFY
 0.0    53554
 1.0    50143
 2.0     4140
 3.0     1244
-1.0      905
Name: count, dtype: int64


0.0 → Muy satisfecho/a

1.0 → Satisfecho/a

2.0 → Insatisfecho/a

3.0 → Muy insatisfecho/a

-1.0 → No sabe / No responde

En total, más del 86% de la muestra se declara satisfecha o muy satisfecha con su vida, lo cual es un indicador muy positivo de bienestar general.

PRUEBA VIH

In [40]:
df.loc[df["_AIDTST4"] == 2, "_AIDTST4"] = 0
df.loc[df["_AIDTST4"] == 9, "_AIDTST4"] = -1

df['_AIDTST4'].value_counts()

/tmp/ipykernel_5781/1639649441.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df.loc[df["_AIDTST4"] == 2, "_AIDTST4"] = 0
/tmp/ipykernel_5781/1639649441.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df.loc[df["_AIDTST4"] == 9, "_AIDTST4"] = -1


_AIDTST4
 0.0    69486
 1.0    35087
-1.0     5413
Name: count, dtype: int64

1.0 → Sí se ha hecho una prueba de VIH

0.0 → No se ha hecho la prueba

-1.0 → No sabe / no responde / dato faltante

Solo 3 de cada 10 personas reportan haberse hecho una prueba de VIH, lo cual puede reflejar:

Falta de acceso a servicios de salud preventiva, estigma o desconocimiento sobre la importancia del testeo o baja percepción de riesgo.

El grupo que nunca se ha testeado (61%) es amplio.

SITUACION MARITAL

In [41]:
df["MARITAL"] = df["MARITAL"] - 1 
df.loc[df["MARITAL"] == 8, "MARITAL"] = -1
df['MARITAL'].value_counts()

/tmp/ipykernel_5781/1867271873.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df["MARITAL"] = df["MARITAL"] - 1


MARITAL
 0.0    65631
 1.0    18644
 2.0    11224
 4.0     8686
 5.0     2954
 3.0     2270
-1.0      577
Name: count, dtype: int64

-  **Casado/a** (0)	65,631 - 55.5%.
-  **Divorciado/a** (1)	18,644 - 15.8%.
-  **Viudo/a** (2)	11,224	- 9.5%.
-  **Nunca se casó** (4)	8,686	- 7.3%.
-  **En pareja no legal** (5)	2,954	- 2.5%.
-  **Separado/a** (3)	2,270 - 1.9%.
-  **Sin respuesta** (-1)	577	- 0.5%.

Más de la mitad de la muestra está casada, lo cual podría correlacionarse con mayor estabilidad emocional, apoyo social y mejor salud física o mental.

Grupo de riesgo emocional: El 25% del total entre divorciados y viudos podría presentar mayor prevalencia de soledad, aislamiento o dificultades psicosociales.

Personas sin pareja: “Nunca casado” y “pareja no legal” suman casi un 10%. Quizás grupos más jóvenes o con estilos de vida distintos.

Separados (3): Aunque representan un grupo pequeño, podrían comportarse de forma distinta respecto al bienestar subjetivo y hábitos de salud.

SITUACION DE VIVIENDA

In [42]:
df["RENTHOM1"] = df["RENTHOM1"] - 1 
df.loc[df["RENTHOM1"] > 5, "RENTHOM1"] = -1
df['RENTHOM1'].value_counts()

/tmp/ipykernel_5781/3779948866.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df["RENTHOM1"] = df["RENTHOM1"] - 1


RENTHOM1
 0.0    87423
 1.0    18868
 2.0     3121
-1.0      574
Name: count, dtype: int64

*Propietarios 0* (76.5%): suelen tener mayor estabilidad residencial, lo que puede correlacionarse con mejor salud mental, menor estrés financiero y mayor satisfacción vital.

*Inquilinos 1* (16.5%): podrían estar más expuestos a inseguridad habitacional, especialmente si se cruzan con bajos ingresos, aislamiento o falta de apoyo emocional.

*Otros arreglos 2*  (2.7%): este grupo puede incluir personas en situación de dependencia, vulnerabilidad o informalidad habitacional.

*No sabe / negado -1* (0.5%): representa una fracción muy pequeña.

SERVICIO MILITAR 

In [43]:
df.loc[df["VETERAN3"] == 2, "VETERAN3"] = 0 
df.loc[df["VETERAN3"] > 3, "VETERAN3"] = -1
df['VETERAN3'].value_counts()

VETERAN3
 0.0    93952
 1.0    15884
-1.0      150
Name: count, dtype: int64

1.0 → Sí ha servido en el ejército.

0.0 → No ha servido.

-1.0 → No sabe / no responde / dato faltante.

Casi 1 de cada 7 personas ha servido en el ejército, lo cual es una proporción significativa. Este grupo puede tener características particulares en salud física, mental o acceso a servicios.

Ser veterano puede estar asociado a:
Mayor prevalencia de enfermedades crónicas o dolor musculares, ser mayor de edad o tener diferencias en acceso a atención médica o beneficios sociales

SITUACION LABORAL 

In [44]:
df["EMPLOY1"] = df["EMPLOY1"] - 1 
df.loc[df["EMPLOY1"] == 8, "EMPLOY1"] = -1

df['EMPLOY1'].value_counts()

/tmp/ipykernel_5781/953924729.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df["EMPLOY1"] = df["EMPLOY1"] - 1


EMPLOY1
 0.0    42203
 6.0    40510
 1.0    11165
 7.0     8450
 4.0     3425
 2.0     1999
 3.0     1422
-1.0      558
 5.0      254
Name: count, dtype: int64

- *Empleado por salario (0)*	42,203	37.9%.
- *Estudiante (6)*	40,510	36.4%.
- *Autónomo (1)*	11,165	10.0%.
- *Jubilado/a (7)*	8,450	7.6%.
- *Desempleado <1 año (4)*	3,425	3.1%.
- *Desempleado ≥1 año (2)*	1,999	1.8%.
- *Ama de casa (3)*	1,422	1.3%.
- *No responde (-1)*	558	0.5%.
- *Incapaz de trabajar (5)*	254	0.2%.
   
Alta proporción de estudiantes (36%): Esto sugiere que tu muestra puede estar sesgada hacia población joven o universitaria. 

Casi 4 de cada 10 personas están empleadas por salario, lo cual es típico en muestras generales.

El grupo de jubilados y personas incapaces de trabajar suma un 8%.

El desempleo (≥1 año o <1 año) representa un 5%, lo cual puede estar asociado a mayor estrés, aislamiento o menor acceso a salud.

EDAD 

In [45]:
df["_AGEG5YR"] = df["_AGEG5YR"] - 1 
df.loc[df["_AGEG5YR"] ==  13, "_AGEG5YR"] = -1
df['_AGEG5YR'].value_counts()

/tmp/ipykernel_5781/2724553984.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df["_AGEG5YR"] = df["_AGEG5YR"] - 1


_AGEG5YR
 8.0     17934
 9.0     17348
 7.0     15774
 6.0     14913
 10.0    13966
 5.0     12871
 11.0     8895
 12.0     6634
-1.0      1651
Name: count, dtype: int64

- 18–24	0.0	
- 25–29	1.0	
- 30–34	2.0	
- 35–39	3.0	
- 40–44	4.0
- 45–49	5.0
- 50–54	6.0	
- 55–59	7.0
- 60–64	8.0
- 65–69	9.0	
- 70–74	10.0	
- 75–79	11.0
- 80 o más	12.0
- No responde	-1.0.

Mayoría adulta mayor: la muestra tiene una gran concentración de personas de 50 años o más, en especial los grupos de 55 a 74 años, lo cual puede explicar la alta prevalencia de enfermedades crónicas, artritis o dificultades funcionales que observamos antes. El grupo de 55–64 años que tiene 33,000 personas, y el de 65–79 años suma más de 29,000 personas ideal para analizar dependencia funcional o cobertura de salud.
Poca representación juvenil entre 18–34 años. 

NIVEL DE INGRESOS

In [46]:
df["_INCOMG1"] = df["_INCOMG1"] - 1 
df.loc[df["_INCOMG1"] >= 7, "_INCOMG1"] = -1
df['_INCOMG1'].value_counts()

/tmp/ipykernel_5781/2176923739.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df["_INCOMG1"] = df["_INCOMG1"] - 1


_INCOMG1
 4.0    29155
 5.0    21371
-1.0    16302
 3.0    11280
 2.0     9878
 1.0     8662
 6.0     7314
 0.0     6024
Name: count, dtype: int64

- Menos de $15k (0.0)
- $15k–$25k (1.0)
- $25k–$35k (2.0)
- $35k–$50k (3.0)
- $50k–$100k (4.0)
- $100k–$200k (5.0)
- $200k o más (6.0)
- No responde (-1.0)

Más del 44% de la muestra tiene ingresos superiores a $50,000 anuales, lo cual sugiere un perfil socioeconómico medio-alto. Sin embargo, también hay un grupo vulnerable (13%) con ingresos menores a $25,000. **Desigualdad interna:** Aunque hay una mayoría con ingresos medios-altos, el grupo de bajos ingresos es relevante y puede presentar mayor prevalencia de, mala salud física o mental, falta de acceso a servicios médicos, aislamiento o falta de apoyo emocional.

NIVEL EDUCATIVO

In [47]:
df["_EDUCAG"] = df["_EDUCAG"] - 1 
df.loc[df["_EDUCAG"] > 5, "_EDUCAG"] = -1

df['_EDUCAG'].value_counts()

/tmp/ipykernel_5781/4019845351.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df["_EDUCAG"] = df["_EDUCAG"] - 1


_EDUCAG
 3.0    49552
 2.0    30316
 1.0    23713
 0.0     6120
-1.0      285
Name: count, dtype: int64

- Graduado de universidad (3)
- Asistencia universitaria (2)
- Secundaria completa (1)
- No terminó secundaria (0)	
- No responde / inválido (-1)

Más del 70% de la muestra tiene al menos educación postsecundaria, lo cual sugiere un nivel socioeducativo medio-alto.

Sólo el 5% no terminó la escuela secundaria, lo que sugiere buena cobertura educativa en la población que estás estudiando. Esta variable es clave para entender diferencias en:

Acceso y uso de servicios de salud, bienestar subjetivo y salud mental, condiciones crónicas como artritis o enfermedad cardíaca.

NUMEROS DE NIÑOS EN EL HOGAR

In [48]:
df["_CHLDCNT"] = df["_CHLDCNT"] - 1 
df.loc[df["_CHLDCNT"] > 6, "_CHLDCNT"] = -1
df['_CHLDCNT'].value_counts()

/tmp/ipykernel_5781/2468622656.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df["_CHLDCNT"] = df["_CHLDCNT"] - 1


_CHLDCNT
 0.0    90787
 1.0     9982
 2.0     5784
 3.0     1868
-1.0      710
 4.0      550
 5.0      305
Name: count, dtype: int64

- 0.0 → Ningún niño
- 1.0	1 niño
- 2.0	2 niños
- 3.0	3 niños
- 4.0	4 niños
- 5.0	5 niños
- -1.0 → No sabe / ausente.

Casi 4 de cada 5 hogares no tienen niños. Los hogares con más de 2 niños son relativamente pocos.
Este patrón sugiere una población adulta sin menores a cargo, lo cual puede estar influido por, 
Edad promedio elevada en la muestra.
Estado civil muchos viudos y divorciados.
Nivel educativo alto y acceso económico.

PERSONAS DE 18 AÑOS VIVIENDO EN CASA

In [49]:
#Se decide conservar estos casos por la posible presencia de personas que viven en refugios, 
# residencias de ancianos u otras situaciones especiales de vivienda.
df['HHADULT'] = df['HHADULT'].astype(int)
df.loc[df["HHADULT"].isin([77,99]), "HHADULT"] = -1
df = df[df["HHADULT"] <= 20]
df['HHADULT'].value_counts()


/tmp/ipykernel_5781/315867495.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df['HHADULT'] = df['HHADULT'].astype(int)


HHADULT
 2     55799
 1     29308
 3     15238
 4      6258
 5      1752
-1       832
 6       474
 7       142
 8        56
 10       27
 9        25
 12       20
 13       16
 11       10
 14        5
 18        5
 15        4
 16        4
 17        3
 0         1
Name: count, dtype: int64

- Hogares de 1 adulto (~25%): posiblemente personas solas.
- Hogares de 2 adultos (~48%): mayoría clásica, probablemente parejas.
- Hogares grandes con mas 4+ adultos, aunque son minoría, podrían corresponder a familias extendidas, situaciones de vulnerabilidad, hogares compartidos odependencia económica.

SEXO

In [50]:
df["SEXVAR"] = df["SEXVAR"] - 1
df['SEXVAR'].value_counts()

SEXVAR
0.0    55123
1.0    54856
Name: count, dtype: int64

-  *Sexo*
-  Hombre: ~50.1%
-  Mujer: ~49.9%

Prácticamente una distribución equilibrada por sexo, lo cual nos es ideal para el análisis comparativo de salud, bienestar o acceso a servicios. Esto nos permite estudiar desigualdades sin sesgo de representación.

AYUDA MEDICA

In [51]:
# Se pasa el 2 a 0, Ya que no cumple la condición.

df.loc[df["MEDCOST1"] > 6, "MEDCOST1"] = -1
df.loc[df["MEDCOST1"] == 2, "MEDCOST1"] = 0

df['MEDCOST1'].value_counts()

MEDCOST1
 0.0    102259
 1.0      7445
-1.0       275
Name: count, dtype: int64

- No (0.0) un 92.2%
- Sí (1.0) un 6.7%
- Sin respuesta (-1.0) un 0.2%.

Casi 1 de cada 15 personas reporta haber necesitado atención médica sin poder pagarla, lo que refleja barreras económicas críticas, incluso en contextos con sistemas de salud público/mixto.

SEGURO DE SALUD

In [52]:
# Se pasa el 2 a 0, Ya que no cumple la condición.

df.loc[df["_HLTHPLN"] > 6, "_HLTHPLN"] = -1
df.loc[df["_HLTHPLN"] == 2, "_HLTHPLN"] = 0

df['_HLTHPLN'].value_counts()

_HLTHPLN
 1.0    103323
 0.0      4029
-1.0      2627
Name: count, dtype: int64

- Sí (1.0)	un 90.6%
- No (0.0)	un 3.5%
- No responde (-1.0) un 2.3%.

Solo un 3.5% de la muestra está sin cobertura médica, lo cual podría parecer bajo, pero esas personas pueden tener un alto riesgo de exclusión del sistema de salud, especialmente si también enfrentan barreras económicas.


PERDIDA DE EMPLEO O REDUCCION DE HORAS

In [53]:
# Variable: SDHEMPLY
# ¿En los últimos 12 meses ha perdido su empleo o le han reducido sus horas de trabajo?
# Cambio de codificacion:
# 1: si     0: no       -1: no contesta
df.loc[df["SDHEMPLY"] > 2, "SDHEMPLY"] = -1
df.loc[df["SDHEMPLY"] == 2, "SDHEMPLY"] = 0
print(df["SDHEMPLY"].value_counts())

SDHEMPLY
 0.0    100725
 1.0      8395
-1.0       859
Name: count, dtype: int64


- No (0) un 91.2%
- Sí (1) un 7.6%
- Sin respuesta (-1.0) un 0.8%
Casi 1 de cada 13 personas o sus convivientes ha sufrido una reducción de ingresos laborales. Este subgrupo es altamente sensible, puede enfrentar pérdida de ingresos, estrés emocional, problemas de cobertura médica y mayor riesgo de aislamiento.

ESCASES DE ALIMENTOS

In [54]:
# Variable: SDHFOOD1
# Durante los últimos 12 meses, ¿con qué frecuencia no te duró la comida que compraste y no tuviste dinero para comprar más? ¿Fue eso…?
# Cambio de codificacion:
# -1: no contesta       [El resto se mantiene tal cual]
df.loc[df["SDHFOOD1"] > 5, "SDHFOOD1"] = -1
print(df["SDHFOOD1"].value_counts())

SDHFOOD1
 5.0    91105
 4.0     7551
 3.0     6895
 1.0     2116
 2.0     1820
-1.0      492
Name: count, dtype: int64


- Nunca (5.0) ~77.7%
- Raramente (4.0) ~6.4%
- A veces (3.0)	~5.9%
- Siempre (1.0)	~1.8%
- Usualmente (2.0)	~1.6%
- No responde (-1.0) ~0.4%

En total, un 9.3% de la muestra presenta inseguridad alimentaria moderada o severa, lo cual es socialmente relevante. 1 de cada 11 hogares vive algún grado de inseguridad alimentaria real, con un 3.4% en niveles altos (1 o 2).

Este dato puede deberse a factores como: Pobreza o desempleo, incapacidad de trabajar, hogares monoparentales o grandes, falta de apoyo institucional o problemas para pagar.

CORTO DE SERVICIOS

In [55]:
# Variable: SDHUTILS
# Durante los últimos 12 meses, ¿hubo alguna ocasión en que una compañía de electricidad, gas, petróleo o agua amenazó 
# con cortar los servicios?
# Cambio de codificacion:
# 1: si     0: no       -1: no contesta
df.loc[df["SDHUTILS"] > 2, "SDHUTILS"] = -1
df.loc[df["SDHUTILS"] == 2, "SDHUTILS"] = 0
print(df["SDHUTILS"].value_counts())

SDHUTILS
 0.0    103605
 1.0      6089
-1.0       285
Name: count, dtype: int64


- No (0.0) ~93.4%
- Sí (1.0) ~5.5%
- Sin respuesta (-1.0) 0.3%

Un 5.5% de la muestra fue amenazada con corte de servicios. Este grupo puede estar en una situación de alta fragilidad financiera. Son personas que seguro no pudieron pagar atención médica, ingresos bajos, desempleo o incapacidad de trabajar, inseguridad alimentaria, hogares con muchas personas a cargo y pocos ingresos o familias monoparentales o sin red de apoyo emocional.


FALTA DE PAGOS

In [56]:
# Variable: SDHBILLS
# Durante los últimos 12 meses, ¿hubo algún momento en el que no pudo pagar su hipoteca, alquiler o facturas de servicios públicos?
# Cambio de codificacion:
# 1: si     0: no       -1: no contesta
df.loc[df["SDHBILLS"] > 2, "SDHBILLS"] = -1
df.loc[df["SDHBILLS"] == 2, "SDHBILLS"] = 0
print(df["SDHBILLS"].value_counts())

SDHBILLS
 0.0    100651
 1.0      8953
-1.0       375
Name: count, dtype: int64


- No tuvo problemas (0.0)	un 91.4%
- Sí tuvo dificultades de pago (1.0)	un 8.1%
- No responde / dato faltante (-1.0)	un 0.3%

Casi 1 de cada 12 personas reportó no poder cubrir su vivienda o servicios básicos en algún momento del último año, una señal de fragilidad económica seria.



FALTA DE TRANSPORTE 

In [57]:
# Variable: SDHTRNSP
# Durante los últimos 12 meses, ¿la falta de transporte confiable le ha impedido asistir a citas médicas, reuniones, 
# trabajar o conseguir las cosas que necesita para la vida diaria?
# Cambio de codificacion:
# 1: si     0: no       -1: no contesta
df.loc[df["SDHTRNSP"] > 2, "SDHTRNSP"] = -1
df.loc[df["SDHTRNSP"] == 2, "SDHTRNSP"] = 0
print(df["SDHTRNSP"].value_counts())

SDHTRNSP
 0.0    103560
 1.0      6176
-1.0       243
Name: count, dtype: int64


- No tuvo dificultades (0.0) un 93.5%
- Sí tuvo dificultades (1.0)	un 5.6%
- Sin respuesta / desconocido (-1.0) un 0.2%

Casi 1 de cada 18 personas o hogares ha tenido problemas para llegar a tiempo por falta de transporte una señal importante de inequidad territorial, movilidad o infraestructura.
En zonas rurales, puede representar distancias excesivas o falta de transporte público.
En grandes ciudades, puede reflejar tiempos de espera, inseguridad, exclusión digital o logística.


HORAS DE SUEÑO

In [58]:
# Variable: SLEPTIM1
# En promedio, ¿cuántas horas duermes en un período de 24 horas?
# Cambio de codificacion:
# -1: no contesta       [El resto se mantiene tal cual]
df.loc[df["SLEPTIM1"] > 24, "SLEPTIM1"] = -1
print(df["SLEPTIM1"].value_counts())

SLEPTIM1
 7.0     33811
 8.0     31788
 6.0     22853
 5.0      6936
 9.0      5487
 4.0      3003
 10.0     2551
-1.0       979
 3.0       775
 12.0      705
 2.0       321
 1.0       229
 11.0      166
 15.0       86
 14.0       76
 16.0       75
 18.0       48
 13.0       39
 20.0       27
 24.0       10
 17.0        6
 19.0        4
 23.0        2
 22.0        2
Name: count, dtype: int64


- 7 h	33,811	~29.2%
- 8 h	31,788	~27.4%
- 6 h	22,853	~19.7%
- 5 h o menos	~10,500	~9.1%
- 9–10 h	~8,000	~6.8%
- Más de 10 h	~470	~0.4%
- Sin respuesta (-1)	979	~0.8%

Más del 56% duerme entre 7 y 8 horas, que es el rango recomendado por la Organización Mundial de la Salud (OMS). 

- Sueño insuficiente menos de 6h, afecta memoria, estado de ánimo, inmunidad y riesgo de enfermedades crónicas.

- Sueño excesivo mas de 9h, puede estar vinculado a depresión, enfermedad crónica, sedentarismo o desempleo.

PRUEBA DE VIH 

In [59]:
# Variable: HIVRISK5
# Voy a leerte una lista. Cuando termine, por favor, dime si alguna de estas situaciones te aplica. No es necesario 
# que me digas cuál. Te has inyectado alguna droga distinta a la recetada en el último año. Has recibido tratamiento 
# por una enfermedad de transmisión sexual o ETS en el último año. Has dado o recibido dinero o drogas a cambio de sexo en el último año.
# Cambio de codificacion:
# 1: si     0: no       -1: no contesta
df.loc[df["HIVRISK5"] > 2, "HIVRISK5"] = -1
df.loc[df["HIVRISK5"] == 2, "HIVRISK5"] = 0
print(df["HIVRISK5"].value_counts())

HIVRISK5
 0.0    107051
 1.0      2542
-1.0       386
Name: count, dtype: int64


- No (0.0) un 96.5%
- Sí (1.0)	un 2.3%
- Sin respuesta (-1.0) un 0.3%

Solo un 2.3% de la muestra reporta haber estado en situación de riesgo de VIH, lo cual puede estar subestimado por estigma social o miedo a declarar conductas sensibles, falta de conciencia sobre qué se considera “riesgo” o falta de seguro.

ACTIVIDAD FISICA DURANTE LOS ULTIMOS 30 DIAS 

In [60]:
# Variable: _TOTINDA
# Adultos que informaron haber realizado actividad física o ejercicio durante los últimos 30 días aparte de su trabajo habitual
# Cambio de codificacion:
# 1: si     0: no       -1: no contesta
df.loc[df["_TOTINDA"] > 2, "_TOTINDA"] = -1
df.loc[df["_TOTINDA"] == 2, "_TOTINDA"] = 0
print(df["_TOTINDA"].value_counts())

_TOTINDA
 1.0    83417
 0.0    26368
-1.0      194
Name: count, dtype: int64


- Sí (1.0)	un 75.5%
- No (0.0) un 23.9%
- Sin respuesta	un 0.2%

Más de 3 de cada 4 personas reportan haber realizado algún tipo de ejercicio, lo cual es una buena señal en términos generales de promoción de salud. El 23.9% que no realizó actividad física puede ser un grupo de riesgo para múltiples condiciones como enfermedad cardíaca, obesidad, salud mental deteriorada, dolor articular o dificultad funcional.

DERRAME CEREBRAL 

In [61]:
# Variable: CVDSTRK3
# ¿Alguna vez te han dicho que tuviste un derrame cerebral?
# Cambio de codificacion:
# 1: si     0: no       -1: no contesta
df.loc[df["CVDSTRK3"] > 2, "CVDSTRK3"] = -1
df.loc[df["CVDSTRK3"] == 2, "CVDSTRK3"] = 0
print(df["CVDSTRK3"].value_counts())

CVDSTRK3
 0.0    104167
 1.0      5563
-1.0       249
Name: count, dtype: int64


- No un 94.8%
- Sí un 5.1%
- No sabe -1 un 0.2%

1 de cada 20 personas ha tenido un derrame cerebral, lo cual es una prevalencia significativa, especialmente porque la muestra incluye muchos adultos mayores.

DIABETES 

In [62]:
# Variable: DIABETE4
# ¿Alguna vez le dijeron que tenía diabetes? (Si la respuesta es «Sí» y la encuestada es mujer, pregunte 
# «¿Fue solo cuando estaba embarazada?». Si la encuestada dice prediabetes o diabetes limítrofe, use el código de respuesta 4).
# Cambio de codificacion:
# 0: No     1: Prediabetes      2: En embarazo      3: No diabetes      -1: No contesta
df.loc[df["DIABETE4"] == 3, "DIABETE4"] = 0
df.loc[df["DIABETE4"] == 1, "DIABETE4"] = 3
df.loc[df["DIABETE4"] == 4, "DIABETE4"] = 1
df.loc[df["DIABETE4"] > 4, "DIABETE4"] = -1
print(df["DIABETE4"].value_counts())

DIABETE4
 0.0    87157
 3.0    18927
 1.0     3072
 2.0      689
-1.0      134
Name: count, dtype: int64


- No tiene diabetes (3.0) un 76.3%
- Prediabetes (4.0) un 16.6%
- Sí, diabetes diagnosticada (1.0) un 2.7%
- Diabetes gestacional (2.0) un 0.6%
- Sin respuesta / inválido (-1.0) un 0.1%

Casi 1 de cada 5 personas tiene prediabetes, y un 3.3% ha tenido diabetes diagnosticada (incluyendo gestacional). Diabetes diagnosticada (3.3%) puede estar asociada a personas de mayor edad, obesidad, enfermedades cardíacas o dificultades funcionales.

In [63]:
columnas = df.columns
columnas
for col in columnas:
    df = df[df[col] != -1]

In [64]:
df.to_csv("../data/processed/datafinal.csv", index=False)